### PR TITLE
SoundWire/ASoC: make Intel link_slaves_found() usable by other SOC vendors

### PR DESCRIPTION
**Pull request by plbossart**
Suggested in https://github.com/orgs/thesofproject/discussions/7882

Completely untested for now, just example code 

@vijendarmukunda please comment and try to see if it fits your needs.

### CODE REVIEW

**Comment by vijendarmukunda** [comment]
@plbossart: Thanks for quick help. We will check and update you our test results.

**Comment by vijendarmukunda** [comment]
verified changes with our SoundWire stack. It's working fine.

**Review by bardliao — DISMISSED** [review]

File: sound/soc/sof/intel/hda.c [hunk 1]
```diff
@@ -1523,19 +1521,16 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
...
-				dev_dbg(bus->dev, "part %x reported %d expected %d on link %d, skipping\n",
+				dev_dbg(dev, "part %x reported %d expected %d on link %d, skipping\n",
 					part_id, reported_part_count, expected_part_count, link_id);
 			}
 		}
 		if (j == num_slaves) {
-			dev_dbg(bus->dev,
-				"Slave %x not found\n",
-				part_id);
+			dev_dbg(dev, "Slave %x not found\n", part_id);
```
Comment (RanderWang) [inline sound/soc/sof/intel/hda.c]: %#x ?

File: sound/soc/sof/intel/hda.c [hunk 2]
```diff
@@ -1523,19 +1521,16 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 				unique_id = SDW_UNIQUE_ID(adr);
 				if (reported_part_count == 1 ||
 				    ids[j].id.unique_id == unique_id) {
-					dev_dbg(bus->dev, "found %x at link %d\n",
-						part_id, link_id);
+					dev_dbg(dev, "found %x at link %d\n", part_id, link_id);
```
Comment (RanderWang) [inline sound/soc/sof/intel/hda.c]: %#x ?

File: sound/soc/sof/intel/hda.c [hunk 3]
```diff
@@ -1523,19 +1521,16 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
...
 					break;
 				}
 			} else {
-				dev_dbg(bus->dev, "part %x reported %d expected %d on link %d, skipping\n",
+				dev_dbg(dev, "part %x reported %d expected %d on link %d, skipping\n",
```
Comment (RanderWang) [inline sound/soc/sof/intel/hda.c]: %#x ?

**Review by RanderWang — COMMENTED** [review]
LGTM.

**Review by RanderWang — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Comment by vijendarmukunda** [comment]
@plbossart :  Could you please push this patch to ASoC tree?.

**Pull request merged**